pkg/internal/appolly/appolly.go

```diff
@@ -87,28 +97,41 @@ func (i *Instrumenter) FindAndInstrument() (<-chan struct{}, error) {
 			}
 		}
 	}()
+
 	// TODO: wait until all the resources have been freed/unmounted
-	return done, nil
+	return nil
 }
 
 // ReadAndForward keeps listening for traces in the BPF map, then reads,
 // processes and forwards them
-func (i *Instrumenter) ReadAndForward() error {
+func (i *Instrumenter) ReadAndForward() {
 	log := log()
 	log.Debug("creating instrumentation pipeline")
 
-	bp, err := pipe.Build(i.ctx, i.config, i.ctxInfo, i.tracesInput)
-	if err != nil {
-		return fmt.Errorf("can't instantiate instrumentation pipeline: %w", err)
-	}
-
 	log.Info("Starting main node")
 
-	bp.Run(i.ctx)
+	i.bp.Run(i.ctx)
 
 	log.Info("exiting auto-instrumenter")
+}
 
-	return nil
+func (i *Instrumenter) Stop() error {
```

Comment:
What's the reason to move the stop/wait code to a function that needs to be explicitly invoked?

Maybe it's just a personal taste, but the previous version looked more "self-contained" as it was a single function invocation.